test/validation/api/crypto/test_vectors.h

```diff
@@ -2527,7 +2527,129 @@ static crypto_test_reference_t zuc_eea3_reference[] = {
 				0xdf, 0x5a, 0x47, 0x3a, 0x57, 0xa4, 0xa0, 0x0d,
 				0x98, 0x5e, 0xba, 0xd8, 0x80, 0xd6, 0xf2, 0x38,
 				0x64, 0xa0, 0x7b, 0x01 }
-	}
+	},
+	{
+		.cipher_key_length = ZUC_EEA3_256_KEY_LEN,
```

Comment:
I am not sure if it is correct to support ZUC-256 through ODP_CIPHER_ALG_ZUC_EEA3 and  ODP_AUTH_ALG_ZUC_EIA3. The API spec says the algorithms are "ZUC based 128-EEA2" and "ZUC-based 128-EIA3", respectively.

Since the current API spec does not describe what the 256-bit variants of the algorithms really are, it is questionable if we can have test vectors for them.

So maybe an API change is needed, perhaps adding new algorithm names as EEA3 and EIA3 may not be appropriate.

Reply:
> The API spec says the algorithms are "ZUC based 128-EEA2" and "ZUC-based 128-EIA3", respectively.

Would updating the API spec like "ZUC based 128/256 EEA3" address your concern?

Reply:
Is there such a thing as "ZUC based 256 EEA3"  (or probably more correctly, ZUC-256 based EEA3)? What would be the defining document of the algorithm? ZUC-256 exists and 128-EEA3 using 128 bit ZUC exists, but I wonder if ZUC-256-EEA3 has a well defined meaning?

Reply:
@JannePeltonen Which one is supported by ODP? Is it ZUC-128 or 128-EEA3?

Reply:
128-EEA3 according to the API. But the API also says that IV should be formatted in a way specified in the EEA3 specification (bearer, direction etc), making one think that maybe the algorithm is really just ZUC and could be used for other contexts than EEA3 by selecting the IV freely (assuming it would be ok by the API, which depends on how seriously one takes the word "should" ).

I am not very familiar with ZUC, EEA3 and EIA3, but perhaps the API could provide simply ZUC instead of EEA3. Then adding ZUC-256 would not be a problem. But ZUC-256 based EIA3 (if it is a thing) could not be implemented efficiently by the application by just ZUC-256 provided by ODP.

Maybe it would be ok to have "ZUC based EEA & EIA" with both 128 and 256 bit keys in the API, but with some clarification on what the ZUC-256 based variants really do (e.g. regarding the IV setup)?

Reply:
> making one think that maybe the algorithm is really just ZUC and could be used for other contexts than EEA3 by selecting the IV freely
That's our impression as well. 

> but perhaps the API could provide simply ZUC instead of EEA3.
We can update the description to state that ODP API is simply ZUC and for EEA3, IV need to be prepared in the described manner.

>  ZUC-256 based variants really do (e.g. regarding the IV setup)
At this point, we don't have enough information on how IV setup would be done. ZUC-256 is a Nokia requirement. Do you have any docs that we can refer to state how IV generation would be done in case of EEA3 based on ZUC-256. If not, we can update documentation such that we will state IV generation for 128-EEA3 and leave rest to application's choice.

Reply:
Please take a look at PR 1664 that adds a mention of ZUC-256 as an experimental algorithm that may be supported. If the API addition makes sense, then this validation test could be added as it is now.